### DISCUSSION TITLE
How can I get layer information in image scan report?

### DISCUSSION THREAD

**Discussion by harpaldhillon** [aquasecurity/trivy#7460] [Q&A]
### Question

Hi All,

I need image scan report where CVE's will also include the layer information. If CVE was present in base image layer or any other layer. Is there a way to find this information easily?

I know we can get "Layer.DiffID" but how can this be mapped to base image layer.

Many thanks.

### Target

Container Image

### Scanner

Vulnerability

### Output Format

JSON

### Mode

None

### Version

```bash
Version: 0.49.1
Vulnerability DB:
  Version: 2
  UpdatedAt: 2024-09-08 06:11:31.065509557 +0000 UTC
  NextUpdate: 2024-09-08 12:11:31.065509187 +0000 UTC
  DownloadedAt: 2024-09-08 06:26:26.130116148 +0000 UTC
Java DB:
  Version: 1
  UpdatedAt: 2024-07-18 01:16:35.848871555 +0000 UTC
  NextUpdate: 2024-07-21 01:16:35.848871395 +0000 UTC
  DownloadedAt: 2024-07-18 20:46:39.59123348 +0000 UTC
```

**Comment by harpaldhillon** [comment]
I'm able to see Layer.DiffID" in html report.

![Screenshot from 2024-09-08 12-21-17](https://github.com/user-attachments/assets/6dc93d62-b93a-4cef-9bdf-b29127a71b32)

**Comment by DmitriyLewen** [comment]
Hello @harpaldhillon 
Thanks for your interest to Trivy!

We scan the images layer by layer and keep the order as in the images.
You can find the list of layers in `json` format (`Metadata.DiffIDs` field).
In the simple case, the base layer should be the first in this array. For example:
```
➜ cat Dockerfile                            
FROM ubuntu

RUN apt -y update

RUN apt -y install curl        
                                                
➜ trivy -q image -f json 7460 | jq .Metadata
{
  "OS": {
    "Family": "ubuntu",
    "Name": "24.04"
  },
  "ImageID": "sha256:63f3a18368152b02c00d4aab88f9015be49f818aa7325987450ffad6d3963aff",
  "DiffIDs": [
    "sha256:aac3af10edc6612ad169ab9f9267dbe90e421f650a089ddc3f63546d9aacb3ed",
    "sha256:f0ff4b6679d6cba6e8cad457e9bee3c6bd851549af71814f8a756e713af51f88",
    "sha256:66585fadefaca7c02deaa9765ad1806c5b4416868cab7815e3a54bfbe3675c74"
  ],
...
```

Regards, Dmitriy
